src/Sylius/Component/Core/Resolver/ShippingMethodsByZonesAndChannelResolver.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\Core\Resolver;
+
+use Sylius\Component\Addressing\Matcher\ZoneMatcherInterface;
+use Sylius\Component\Core\Model\ChannelInterface;
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\ShipmentInterface;
+use Sylius\Component\Shipping\Model\ShippingSubjectInterface;
+use Sylius\Component\Shipping\Repository\ShippingMethodRepositoryInterface;
+use Sylius\Component\Shipping\Resolver\MethodsResolverInterface;
+
+/**
+ * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ */
+class ShippingMethodsByZonesAndChannelResolver implements MethodsResolverInterface
+{
+    /**
+     * @var ShippingMethodRepositoryInterface
+     */
+    private $shippingMethodRepository;
+
+    /**
+     * @var ZoneMatcherInterface
+     */
+    private $zoneMatcher;
+
+    /**
+     * @param ShippingMethodRepositoryInterface $shippingMethodRepository
+     * @param ZoneMatcherInterface $zoneMatcher
+     */
+    public function __construct(
+        ShippingMethodRepositoryInterface $shippingMethodRepository,
+        ZoneMatcherInterface $zoneMatcher
+    ) {
+        $this->shippingMethodRepository = $shippingMethodRepository;
+        $this->zoneMatcher = $zoneMatcher;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getSupportedMethods(ShippingSubjectInterface $subject)
```

Comment:
That is original name of the method. Maybe we can keep it for now and also suggests what kind of methods we get from this resolver. ;)